zanata-war/src/main/java/org/zanata/action/ProjectPermissionDialog.java

```diff
@@ -0,0 +1,250 @@
+     * @param checked the current checkbox value
+     */
+    public void bindProjectRole(String role, boolean checked) {
+        if(StringUtils.equalsIgnoreCase(role, ProjectRole.TranslationMaintainer.name())) {
+            data.setTranslationMaintainer(checked);
+        } else if(StringUtils.equalsIgnoreCase(role, ProjectRole.Maintainer.name())) {
+            data.setMaintainer(checked);
+        }
+    }
+
+    /**
+     * Update role membership for a locale-specific role based on the current
+     * checkbox value.
+     *
+     * @param localeRole represents both the locale and the role
+     *                   format: {localeId}:{role}. e.g en-US:Reviewer
+     * @param checked the current checkbox value
+     */
+    public void bindTranslationRole(String localeRole, boolean checked) {
+        String[] localeRoleList = StringUtils.split(localeRole, ':');
+        HLocale hLocale = localeServiceImpl.getByLocaleId(localeRoleList[0]);
+        String role = localeRoleList[1];
+
+        for (PersonProjectMemberships.LocaleRoles localeRoles: data.getLocaleRoles()) {
+            if(localeRoles.getLocale().equals(hLocale)) {
+                if (StringUtils.equalsIgnoreCase(role, LocaleRole.Translator.name())) {
+                    localeRoles.setTranslator(checked);
+                } else if (StringUtils.equalsIgnoreCase(role, LocaleRole.Reviewer.name())) {
+                    localeRoles.setReviewer(checked);
+                } else if (StringUtils.equalsIgnoreCase(role, LocaleRole.Coordinator.name())) {
+                    localeRoles.setCoordinator(checked);
+                }
+                return;
+            }
+        }
+
+        // No LocaleRoles for the given locale, so create a new one.
+        List<LocaleRole> roleList = Lists.newArrayList(LocaleRole.valueOf(role));
+        data.addLocaleRoles(hLocale, roleList);
+    }
+
+    /**
+     * Save the permissions selections from permissionDialogData to the database.
+     */
+    public void saveSelections() {
+        if (data == null) {
+            log.error("Tried to save permissionDialogData but it is null");
+            return;
+        }
+
+        project = projectDAO.findById(getProject().getId());
+
+        // Hibernate will have problems working with detached HPerson and HLocale
+        // so they are all attached before any persistence is attempted.
+        HPerson person = personDAO.findById(data.getPerson().getId());
+        data.setPerson(person);
+        for (PersonProjectMemberships.LocaleRoles roles : data.getLocaleRoles()) {
+            roles.setLocale(localeServiceImpl
+                    .getByLocaleId(roles.getLocale().getLocaleId()));
+        }
+
+        final boolean canManageMembers = identity.hasPermission(project, "manage-members");
+        final boolean canManageTransMembers = identity.hasPermission(project, "manage-translation-members");
+        final boolean canChangeAnyMembers = canManageMembers || canManageTransMembers;
```

Comment:
@huangp There is no way to do this in the UI unless you edit the page code - the user has no buttons to open the dialog. Is it worth trying to give feedback for a case that is not allowed in the UI?